pkg/dockerregistry/server/quotarestrictedblobstore.go

```diff
@@ -58,7 +99,7 @@ var _ distribution.BlobStore = &quotaRestrictedBlobStore{}
 func (bs *quotaRestrictedBlobStore) Put(ctx context.Context, mediaType string, p []byte) (distribution.Descriptor, error) {
 	context.GetLogger(ctx).Debug("(*quotaRestrictedBlobStore).Put: starting")
 
-	if err := admitBlobWrite(ctx, bs.repo); err != nil {
+	if err := admitBlobWrite(ctx, bs.repo, int64(len(p))); err != nil {
```

Comment:
This can't actually be holding an entire layer in memory in a byte array at once, can it?